parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -165,6 +185,7 @@ where
 			BuyExecution { fees: xcm_fee, weight_limit: Unlimited },
 			// Fund the snowbridge sovereign with the required deposit for creation.
 			DepositAsset { assets: Definite(deposit.into()), beneficiary: bridge_location },
+			DescendOrigin(X1(PalletInstance(inbound_queue_pallet_index))),
```

Comment:
Shouldn't this `DescendOrigin` be placed at the start of message, before `BuyExecution`?

As otherwise, we have three origins in this message, which makes it more complicated.

cc @alistair-singh

Reply:
@claravanstaden  I agree with @vgeddes  here, better to move `DescendOrigin` to first, revamp accordingly in 
https://github.com/Snowfork/snowbridge/pull/1034/files#diff-9218e396410f01117c1c305b9cf006f6498bb0b3c0aedb8952a8450b4b1c5715

Reply:
Thanks @yrong, I merged it.